### DISCUSSION TITLE
delta_time wrong with Quali

### DISCUSSION THREAD

**Discussion by MatisBruneau** [theOehrly/Fast-F1#130] [Q&A]
Hello,
I was trying to redo the example of the comparison of quali laps that is in the documentation for fastf1.utils.delta_time(reference_lap, compare_lap). However, when I compare LEC and VER, I my graph is wrong because the final gap is not 0.123 seconds, but over 0.3 seconds. The laptime are correct. Where does the error come from ?

**Comment by theOehrly** [comment]
I just checked the docs and there was some issue with that graphic. I've corrected that now. Still, it shouldn't be more than 0.3 seconds but slightly less.

**Reply by MatisBruneau** [reply]
Hi, 
I copy/paste the example code (the one with HAM and LEC at the Emilie-Romagna GP), and I get indeed something below 0.3 s. However, on Wikipedia I can read that the time gap between them was 0.740-0.411 = 0.329 sec. So why the graph does not show this value at the end ?

EDIT: I just found that the two laps' telemetries do not have the same number of rows. I think that this is causing the issue, because the last delta_value seems not to be the delta between the two lap time. Could you force that ?

**Reply by theOehrly** [reply]
Have you read this: https://theoehrly.github.io/Fast-F1/utils.html ?

The sample rate is just very low and the calculation therefore not extremely accurate.

**Reply by MatisBruneau** [reply]
Oh I read it but maybe to quickly, sorry. However I don't understand why the delta_time cannot have as last value the difference between the lap time ? I don't find the precise way of computing it in the documentation you sent. 
Thanks for your precious work! This package is really good for F1 and data science lovers!

**Reply by theOehrly** [reply]
Well, if I just set the correct value as the last value then there will simply be a jump in the data. It doesn't make the rest of the data more accurate.

There is currently no "accurate" way to calculate the delta time.
